### PR TITLE
Correct java -version and -Xinternalversion

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Signed-off-by: Andrew Leonard <andrew_m_leonard@uk.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
Please regenerate the generated-configure scripts to fix the merge conflicts.

File: closed/autoconf/custom-hook.m4 [hunk 1]
```diff
@@ -164,8 +164,18 @@ AC_DEFUN_ONCE([OPENJDK_VERSION_DETAILS],
 [
   OPENJDK_SHA=`git -C $TOPDIR rev-parse --short HEAD`
   OPENJDK_TAG=`git -C $TOPDIR describe --abbrev=0 --tags --match "jdk-10*" "${OPENJDK_SHA}"`
+
+  if test "x$OPENJDK_TAG" = x; then
+    LAST_TAGGED_REVISION=`git -C $TOPDIR rev-list --tags --max-count=1`
+    OPENJDK_TAG=`git -C $TOPDIR describe --abbrev=0 --tags --match "jdk-10*" "${LAST_TAGGED_REVISION}"`
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4]: This should not have been added for earlier versions: it was a hack to workaround an issue with the build flow at AdoptOpenJDK. If they clone the repo (at least enough to include tags) the assignment to `OPENJDK_TAG` above will work.

**Comment by andrew-m-leonard** [comment]
*"Release" build:*
```
openjdk version "10" 2018-03-29
OpenJDK Runtime Environment (build 10+46)
Eclipse OpenJ9 VM (build master-9adecd31, JRE 10 Linux amd64-64 Compressed References 20180329_000000 (JIT enabled, AOT enabled)
OpenJ9   - 9adecd31
OMR      - a05605ea
JCL      - 33c38bfc89 based on jdk-10+46)

java -Xinternalversion
Eclipse OpenJ9 OpenJDK 64-bit Server VM (10+46) from linux-amd64 JRE with Extensions for OpenJDK for Eclipse OpenJ9 10.0.0.0, built on Mar 29 2018 11:46:10 by admin with g++ (Ubuntu 4.8.4-2ubuntu1~14.04.4) 4.8.4
```

*"User" build:*
```
openjdk version "10-internal" 2018-03-20
OpenJDK Runtime Environment (build 10-internal+0-adhoc.admin.jdk10Public)
Eclipse OpenJ9 VM (build master-9adecd31, JRE 10 Linux amd64-64 Compressed References 20180329_000000 (JIT enabled, AOT enabled)
OpenJ9   - 9adecd31
OMR      - a05605ea
JCL      - 33c38bfc89 based on jdk-10+46)

/jdk/bin/java -Xinternalversion
Eclipse OpenJ9 OpenJDK 64-bit Server VM (10-internal+0-adhoc.admin.jdk10Public) from linux-amd64 JRE with Extensions for OpenJDK for Eclipse OpenJ9 10.0.0.0, built on Mar 29 2018 11:53:09 by admin with g++ (Ubuntu 4.8.4-2ubuntu1~14.04.4) 4.8.4
```

**Comment by andrew-m-leonard** [comment]
Personal build: https://hyc-rt-openjdk-jenkins.swg-devops.com/job/build_ibmsdk/291/console

**Comment by andrew-m-leonard** [comment]
@keithc-ca Keith, if you could review and merge please?
thanks

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk10

**Comment by AdamBrousseau** [comment]
Jenkins compile plinux

**Comment by AdamBrousseau** [comment]
Jenkins test sanity plinux jdk10

**Comment by AdamBrousseau** [comment]
(The compile should be fine, the Sanity will likely still fail due to eclipse/openj9#1543. Because of that, you have to request 10 testing specifically for now because I have not included it in the "all" groupings)

**Comment by keithc-ca** [comment]
I restarted the sanity job: it seems a network failure interrupted cloning the repo.

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux jdk10

(the restart didn't go well)

**Comment by keithc-ca** [comment]
Jenkins test sanity plinux jdk10

**Comment by keithc-ca** [comment]
The sanity test encountered https://github.com/eclipse/openj9/issues/1543, but I don't think we need to delay merging this.

**Pull request merged**